wp-job-manager-functions.php

```diff
@@ -723,6 +723,10 @@ function job_manager_upload_file( $file, $args = array() ) {
 		$allowed_mime_types = $args['allowed_mime_types'];
 	}
 
+	if( is_wp_error( $file = apply_filters( 'job_manager_upload_file_pre_upload', $file, $args, $allowed_mime_types ) ) ) {
```

Comment:
@spencerfinnell nah this way by adding one filter we add two use case scenarios ... one for modifying the `$file` data before being uploaded, and the other for returning a `WP_Error` to prevent upload and return that error